applications/luci-app-advanced-reboot/root/usr/share/rpcd/ucode/luci.advanced-reboot

```diff
@@ -0,0 +1,483 @@
+	} else {
+		alt_partition_unmount(op_ubi, ubi_vol);
+	}
+	return { label: null, os: null };
+}
+
+/* Fallback: read label directly from MTD using dd pipeline */
+function get_partition_info_fallback(mtd, offset, vendor) {
+	let label = null;
+	let os = null;
+	let tag = command("dd if=/dev/" + mtd + " bs=1 skip=" + offset + " count=64");
+	if (tag) {
+		let m1 = match(tag, /Linux version ([^ \n]+)/);
+		if (m1 && m1[1]) os = m1[1];
+		else {
+			let m2 = match(tag, /Linux-([0-9.]+)/);
+			if (m2 && m2[1]) os = m2[1];
+		}
+		if (match(tag, /OpenWrt/)) {
+			label = "OpenWrt";
+		} else if (vendor && vendor != "" && match(tag, regexp(vendor))) {
+			label = vendor;
+		} else {
+			label = "Unknown";
+		}
+	} else {
+		label = vendor ? vendor + " (Compressed)" : "Unknown (Compressed)";
+	}
+	return { label: label, os: os };
+}
+
+/* Dual-flag helpers (Zyxel path) */
+function read_dual_flag_mtd() {
+	for (let name in ["0:dual_flag", "0:DUAL_FLAG"]) {
+		let dev = command(". /lib/functions.sh; find_mtd_part " + shellquote(name));
+		if (dev) return dev;
+	}
+	return null;
+}
+
+function read_dual_flag_value(dev) {
+	if (!dev || dev == "" || !file_exists(dev)) return null;
+	let r = command(
+		"dd if=" +
+			shellquote(dev) +
+			" bs=1 count=1 2>/dev/null | hexdump -n 1 -e '" +
+			'1/1 "%02x"' +
+			"'"
+	);
+	return r || null;
+}
+
+function write_dual_flag_value(dev, cur) {
+	cur = "\\x" + cur;
+	let rc = command(
+		"printf %b " +
+			shellquote(cur) +
+			" > " +
+			shellquote(dev) +
+			" 2>/dev/null; echo $?"
+	);
+	return rc == "0";
+}
+
+/* Obtain info: mirrors shell behavior */
+function obtain_device_info() {
+	let board = get_board_name();
+	if (!board) return { error: "NO_BOARD_NAME" };
+
+	let d = find_device_info(board);
+	if (!d) return { error: "NO_BOARD_NAME_MATCH", rom_board_name: board };
+
+	/* parse new-schema device object */
+	let dev = d.device ?? {};
+	let cmds = d.commands ?? {};
+	let parts = d.partitions && type(d.partitions) == "array" ? d.partitions : [];
+	let getcmd = cmds.get ?? "fw_printenv";
+	let active_num = null;
+
+	/* read current values for all params */
+	let curvals = [];
+
+	if (cmds.params && type(cmds.params) == "array" && length(cmds.params) > 0) {
+		for (let i = 0; i < length(cmds.params); i++) {
+			let p = cmds.params[i];
+			let v = null;
+			if (p)
+				v = command(getcmd + " -n " + shellquote(p) + " 2>/dev/null") || null;
+			push(curvals, v);
+		}
+	} else {
+		let df = read_dual_flag_mtd();
+		let v = null;
+		if (!df) return { error: "NO_DUAL_FLAG", rom_board_name: board };
+		if (!file_exists(df))
+			return { error: "NO_DUAL_FLAG_BLOCK", rom_board_name: board };
+		v = read_dual_flag_value(df);
+		push(curvals, v);
+	}
+
+	let out_parts = [];
+	for (let i = 0; i < length(parts); i++) {
+		let p = parts[i] ?? {};
+		let num = p.number != null ? p.number : i;
+		let mtd = p.mtd ?? null;
+		let info = null;
+		let v = null;
+
+		if ("" + curvals[0] == "" + p.param_values[0]) {
```

Comment:
This looks like a logic error. Are you checking length or whether you can successfully concatenate a string?

Reply:
seems to work as intended and only evaluates if curvals[0] matches p.param_values[0] tho.

Reply:
sorry, I see the logic now :D